### PR TITLE
Allow users to specify which packages should be installed during ci setup

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -1741,9 +1751,10 @@ def _load_forge_config(forge_dir, exclusive_config_file, forge_yml=None):
         "build_with_mambabuild": True,
         # feedstock checkout git clone depth, None means keep default, 0 means no limit
         "clone_depth": None,
-        # Specific channel for package can be given with
-        # `${url or channel_alias}::package_name`, defaults to conda-forge channel_alias
-        "remote_ci_setup": "conda-forge-ci-setup=3",
+        "remote_ci_setup": ["conda-forge-ci-setup=3"],
+        # Specific channel for packages for conda-forge-ci-setup,
+        # defaults to conda-forge
+        "remote_ci_setup_channels": ["conda-forge"],
```
Comment (isuruf) [inline conda_smithy/configure_feedstock.py:1757]: Let's remove this as it's not needed.
```suggestion
```

Comment (hmaarrfk) [reply conda_smithy/configure_feedstock.py:1757]: Are you suggesting the user uses `remote_ci_setup` and adds the channels in there?

Seems a little difficult to get right with the escpae characters `<>|`

Comment (hmaarrfk) [reply conda_smithy/configure_feedstock.py]: Thank you for being so explicit! Thank you. Trying to simplify now. WIll see how the test goes.

File: conda_smithy/configure_feedstock.py [hunk 2]
```diff
@@ -1741,9 +1753,7 @@ def _load_forge_config(forge_dir, exclusive_config_file, forge_yml=None):
         "build_with_mambabuild": True,
         # feedstock checkout git clone depth, None means keep default, 0 means no limit
         "clone_depth": None,
-        # Specific channel for package can be given with
-        # `${url or channel_alias}::package_name`, defaults to conda-forge channel_alias
```
Comment (isuruf) [inline conda_smithy/configure_feedstock.py:1745]: Let's keep this comment

Comment (hmaarrfk) [reply conda_smithy/configure_feedstock.py]: Thank you. I added a slightly reformated comment

**Comment by isuruf** [comment]
> @isuruf what I find strange now is that we ask that users specify conda-forge-ci-setup but not conda-build or pip.

That's fine. A use case was to use a custom `myorg-ci-setup` and another was to try `conda-forge-ci-setup=4`.

**Comment by hmaarrfk** [comment]
@xhochy let me know if there is a usecase we forgot as part of this PR. 

You can check sample output in https://github.com/conda-forge/tensorflow-feedstock/pull/209#pullrequestreview-881045040

**Comment by isuruf** [comment]
There are more jinja templates using `remote_ci_setup` like github_actions, appveyor.

**Comment by hmaarrfk** [comment]
I think it got them now... sorry for the rush.

**Comment by jakirkham** [comment]
Would it make sense to add this to the conda-forge docs or the README here as well?

**Pull request merged**